IrohaCrypto/Classes/sr25519/SS58AddressFactory.m

```diff
@@ -17,10 +17,27 @@
 @implementation SS58AddressFactory
 
 - (nullable NSString*)addressFromPublicKey:(id<IRPublicKeyProtocol> _Nonnull)publicKey
-                                      type:(SNAddressType)type
+                                      type:(UInt8)type
                                      error:(NSError*_Nullable*_Nullable)error {
+
+    UInt16 ident = type & 0b0011111111111111;
+
     NSMutableData *addressData = [NSMutableData data];
-    [addressData appendData:[NSData dataWithBytes:&type length:1]];
+
+    switch (ident) {
```

Comment:
Spec says that extended format encodes values between 64 and 16,383. So I think we need to use ```if (ident < 64) { ... } else { ... }```

Reply:
>(128..=255 inclusive) Reserved for future address format extensions.
so it's kind of future proofing